src/ray/core_worker/core_worker.cc

```diff
@@ -2167,19 +2167,26 @@ void CoreWorker::HandleCancelTask(const rpc::CancelTaskRequest &request,
                                   rpc::SendReplyCallback send_reply_callback) {
   absl::MutexLock lock(&mutex_);
   TaskID task_id = TaskID::FromBinary(request.intended_task_id());
-  bool success = main_thread_task_id_ == task_id;
+  bool kill_success = main_thread_task_id_ == task_id;
+  bool success = kill_success;
 
   // Try non-force kill
-  if (success && !request.force_kill()) {
+  if (kill_success && !request.force_kill()) {
     RAY_LOG(INFO) << "Interrupting a running task " << main_thread_task_id_;
-    success = options_.kill_main();
+    kill_success = options_.kill_main();
+  } else if (!kill_success) {
+    
+    // If the task is not currently running, check if it is in the worker's queue of normal tasks, and remove it if found. 
+    if (direct_task_receiver_->CancelQueuedNormalTask(task_id)) {
+      success=true;
+    }
```

Comment:
Can you just assign?  `success = direct_task_receiver_->CancelQueuedNormalTask(task_id)`?